backend/server.js

```diff
@@ -30,82 +38,136 @@ function setupSocketIO(server) {
-      sendBroadcast(clientChannels.AVAILABLE_TABLES, availableTables());
-
-      // send data to client
-      clientSocket.emit(clientChannels.LOGIN_TABLE, {
-        tableNumber: tableNumber,
-        matchStarted: matchStarted
-      });
+      clientLogin(clientSocket, data);
     });
 
-    // event fired when the client sends a message
-    clientSocket.on(clientChannels.SEND_MESSAGE, data => {
-      console.log(`${clientSocket.id} -> ${data}`);
+    // event fired when a start round is triggered
+    clientSocket.on(clientChannels.GET_MATCH, data => {
+      sendMatchToClient(clientSocket, data);
     });
 
     // event fired when a client disconnects, remove it from the list
     clientSocket.on(clientChannels.DISCONNECT, data => {
-      if (addedDevice) {
-        connectedClients.delete(clientSocket.id);
-        console.info(`Client logout [id=${clientSocket.id}]`);
-
-        sendBroadcast(clientChannels.AVAILABLE_TABLES, availableTables());
-      }
-      console.log(`Client gone [id=${clientSocket.id}]`);
+      clientLogout(clientSocket);
     });
   });
 }
 
-function mapHasValue(inputMap, searchedValue) {
-  const values = Array.from(inputMap.entries());
-  return values.some(([_, value]) => value === searchedValue);
-}
-
 function sendStartRoundBroadcast() {
   if (matchStarted) return;
-  
+
   matchStarted = true;
   sendBroadcast(clientChannels.START_ROUND, null);
 }
 
-// this method is used to submit a broadcast event to all clients
-function sendBroadcast(eventName, data) {
-  if (serverSocket) {
-    serverSocket.sockets.emit(eventName, data);
-    console.log(`server emit broadcast: ${eventName}`);
-    console.log(`--- data was ${data}`);
+function clientLogin(clientSocket, data) {
+  const { tableNumber } = data;
+
+  // verify if max amount of connected devices/table is reached
+  if (connectedClients.size === MAX_AMOUNT_TABLE) {
+    clientSocket.emit(clientChannels.LOGIN_ERROR, data);
+    return;
   }
+
+  // verify if a client is already connected to a table
+  if (mapHasValue(connectedClients, tableNumber)) {
+    clientSocket.emit(clientChannels.LOGIN_ERROR, data);
+    return;
+  }
+
+  // login client
+  connectedClients.set(clientSocket.id, tableNumber);
+  console.info(`Client login [id=${clientSocket.id}] [table=${tableNumber}]`);
+
+  // send current server state to clients
+  sendBroadcast(clientChannels.AVAILABLE_TABLES, availableTables());
+
+  // send data to client
+  clientSocket.emit(clientChannels.LOGIN_TABLE, {
+    tableNumber: tableNumber,
+    matchStarted: matchStarted
+  });
+}
+
+function clientLogout(clientSocket) {
+  if (connectedClients.has(clientSocket.id)) {
+    connectedClients.delete(clientSocket.id);
+    console.info(`Client logout [id=${clientSocket.id}]`);
+
+    sendBroadcast(clientChannels.AVAILABLE_TABLES, availableTables());
+  }
+  console.log(`Client gone [id=${clientSocket.id}]`);
+}
+
+function scrubleMatches(players) {
```

Comment:
What do you mean with `scruble`? It's not really a word :)

Reply:
I am Sorry! The word I seeked for was 'dice'